grayskull/strategy/py_build.py

```diff
@@ -0,0 +1,144 @@
+"""
+Use pypa/build to get project metadata from a checkout. Create a recipe suitable
+for inlinining into the first-party project source tree.
+"""
+
+import logging
+import tempfile
+from importlib.metadata import PathDistribution
+from pathlib import Path
+
+import build
+from conda.exceptions import InvalidMatchSpec
+from conda.models.match_spec import MatchSpec
+from packaging.requirements import Requirement
+from souschef.recipe import Recipe
+
+from grayskull.config import Configuration
+from grayskull.strategy.abstract_strategy import AbstractStrategy
+from grayskull.strategy.pypi import compose_test_section
+
+log = logging.getLogger(__name__)
+
+
+class PyBuild(AbstractStrategy):
+    @staticmethod
+    def fetch_data(recipe: Recipe, config: Configuration, sections=None):
+        project = build.ProjectBuilder(config.name)
+
+        recipe["source"]["path"] = "../"
+        # XXX relative to output. "git_url: ../" is also a good choice.
+
+        with tempfile.TemporaryDirectory(prefix="grayskull") as output:
+            build_system_requires = project.build_system_requires
+            requires_for_build = project.get_requires_for_build("wheel")
+            # If those are already installed, we can get the extras requirements
+            # without invoking pip e.g. setuptools_scm[toml]
+            print("Requires for build:", build_system_requires, requires_for_build)
+
+            # build the project's metadata "dist-info" directory
+            metadata_path = Path(project.metadata_path(output_directory=output))
+
+            distribution = PathDistribution(metadata_path)
+
+            # real distribution name not pathname
+            config.name = distribution.name  # see also _normalized_name
+            config.version = distribution.version
+
+            # grayskull thought the name was the path. correct that.
+            if recipe[0] == '#% set name = "." %}':  # XXX fragile
+                # recipe[0] = x does not work
+                recipe._yaml._yaml_get_pre_comment()[0].value = (
+                    f'#% set name = "{config.name}" %}}\n'
+                    f'#% set version = "{config.version}" %}}\n'
+                )
+
+                recipe["package"]["version"] = "{{ version }}"
```

Comment:
This puts quotes around `version: '{{ version }}'`; how do we control the YAML more closely with conda-souschef's Recipe?

Reply:
conda-souschef is just a wrapper around ruamel yaml, you can still manipulate the ruamel yaml object directly.
you can do something like:
```python
recipe["package"]["version"].value = "{{ version }}"
```